### PR TITLE
JBPM-5998 : Tasks: Completed tasks disappear from All tab when proces…

### PR DESCRIPTION
**Pull request by nmirasch**
…s instance completes
Added business admin preprocessor to check is the user who is requesting the query is business administrator.

### CODE REVIEW

**Comment by nmirasch** [comment]
depends on https://github.com/kiegroup/jbpm-wb/pull/806 
@cristianonicolai @bsremac @mswiderski  Could you take a look? thanks!

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessor.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        if (identityProvider == null) {
+            return;
+        }
+        boolean hasBusinessAdminRights = identityProvider.getRoles().stream().filter(s -> s.equals(ADMIN_GROUP)).findFirst().isPresent();
+        if (!hasBusinessAdminRights) {
+            hasBusinessAdminRights = ADMIN_USER.equals(identityProvider.getName());
+        }
+        if (hasBusinessAdminRights) {
+            return;
+        }
```
Comment (bsremac) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessor.java]: How about replacing this logic with a simple OR? Something along the lines of:
```java
        if (ADMIN_USER.equals(identityProvider.getName() || 
            identityProvider.getRoles().stream().filter(s -> s.equals(ADMIN_GROUP)).findFirst().isPresent()) {
                return;
        }
```

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package org.jbpm.kie.services.impl.query.preprocessor;
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java]: missing license header

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class BusinessAdminPreprocessorTest {
+
+    @Mock
+    IdentityProvider identityProvider;
+
+    DataSetLookup dataSetLookup;
+
+    //@InjectMocks
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java]: worth removing if not needed

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+        dataSetLookup= new DataSetLookup();
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+
+        assertEquals("TASKID = -1", dataSetLookup.getFirstFilterOp().getColumnFilterList().get(0).toString());
+    }
+
+    @Test
+    public void testSetUserWithoutRoles() {
```
Comment (bsremac) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java]: This test is a duplicate of the previous test (lines 75-82).
However, I'd encourage you to leave this test as it is and remove lines 75-82 for clarity reasons.

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        String role1 = "admins";
+        String role2 = "role2";
+        String userId = "userId";
+
+        when(identityProvider.getRoles()).thenReturn(Arrays.asList(role1, role2));
+        when(identityProvider.getName()).thenReturn(userId);
+
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+        assertNull(dataSetLookup.getFirstFilterOp());
```
Comment (bsremac) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java]: I would recommend replacing this assertion with adding @Spy annotation to dataSetLookup field and then verifying that there was no interaction with the object i.e. _verifyZeroInteractions(dataSetLookup)_

**Comment by nmirasch** [comment]
@mswiderski @bsremac @cristianonicolai  changes done.

**Comment by nmirasch** [comment]
Changed one of new headers copyright from year 2016 -> 2017

**Comment by bsremac** [comment]
+1

**Comment by mswiderski** [comment]
is this ready to be merged or it wait for anything else?

**Comment by nmirasch** [comment]
this pr seems to be ready but have to be merged at the same time as https://github.com/kiegroup/jbpm-wb/pull/806

**Comment by tsurdilo** [comment]
Jenkins retest please

**Pull request merged**